### PR TITLE
Refactor Code : Assign Goonj Office

### PR DESCRIPTION
**Pull request by belwalshubham**
**Organization Creation (organisationCreated):**
1. When a new organization is created, the system checks if it is an "Institute" subtype.
2. If it's an "Institute," the organization's ID is logged and stored for further processing.

**Assigning Office and Coordinator (setOfficeDetails):**
1. When a primary address is added to the organization, the system looks for a corresponding Goonj office based on the organization's state.
2. If no office is found for the state, a fallback office is assigned.
3. The system then identifies and assigns a relevant coordinator from the state office to manage the organization.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new event handler for organization creation.
	- Added logging for organization details during creation.

- **Improvements**
	- Enhanced validation and logging in the office details setting process.
	- Streamlined chapter group assignment logic with improved checks.

### CODE REVIEW

File: wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php [hunk 1]
```diff
@@ -139,35 +139,48 @@ private static function addContactToGroup($contactId, $groupId) {
...
+    if ($op !== 'create' || $objectName !== 'Organization') {
+      return FALSE;
     }
 
-    if (!($stateField = self::findStateField($params))) {
-      return;
+    \Civi::log()->info('Organization created: ', [
+      'id' => $objectId,
+      'subtypes' => $objectRef->contact_sub_type,
+    ]);
```
Comment (tarunnjoshi) [inline wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php:151]: Why do we need logs here is there any case where this code breaks and you want these logs? I don't think we needed if you want try to use try catch blocks.

Why are you not removing logs, before giving them for review please remove them

File: wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php [hunk 2]
```diff
@@ -139,35 +139,48 @@ private static function addContactToGroup($contactId, $groupId) {
...
+
+    if (empty($subTypes)) {
+      return FALSE;
     }
 
-    $stateId = $stateField['value'];
-    $contactId = $stateField['entity_id'];
+    // The ASCII control character \x01 represents the "Start of Header".
+    // It is used to separate values internally by CiviCRM for multiple subtypes.
+    $subtypes = explode("\x01", $subTypes);
```
Comment (tarunnjoshi) [inline wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php:160]: What this returns can you please show me

File: wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php [hunk 3]
```diff
@@ -139,35 +139,48 @@ private static function addContactToGroup($contactId, $groupId) {
...
-      ->addWhere('id', '=', $contactId)
-      ->execute();
+    self::$organizationId = $objectId;
 
-    if (!$stateId) {
-      \CRM_Core_Error::debug_log_message('Cannot assign Goonj Office to institution id: ' . $contactId);
-      return;
+    \Civi::log()->info('Organization set: ', [
+      'id' => self::$organizationId,
+    ]);
```
Comment (tarunnjoshi) [inline wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php:171]: Please remove unnecessary logs

File: wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php [hunk 4]
```diff
@@ -184,13 +197,18 @@ public static function setOfficeDetails($op, $groupID, $entityID, &$params) {
...
 
-    $updateGoonjOffice = Organization::update(FALSE)
+    Organization::update(FALSE)
       ->addValue('Review.Goonj_Office', $stateOfficeId)
-      ->addWhere('id', '=', $contactId)
+      ->addWhere('id', '=', self::$organizationId)
       ->execute();
 
+    if (!$stateId) {
+      \Civi::log()->info('state not found', ['organizationId' => self::$organizationId, 'stateId' => $stateId]);
```
Comment (tarunnjoshi) [inline wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php:206]: these logs are good, where it only logs the value when there is no stateId, and on above on the code you are printing logs everytime when functions runs

**Review by tarunnjoshi — CHANGES_REQUESTED** [review]

File: wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php [hunk 5]
```diff
@@ -139,35 +139,40 @@ private static function addContactToGroup($contactId, $groupId) {
...
-      return;
+    if ($op !== 'create' || $objectName !== 'Organization') {
+      return FALSE;
     }
 
-    if (!($stateField = self::findStateField($params))) {
-      return;
+    $subTypes = $objectRef->contact_sub_type;
+
+    if (empty($subTypes)) {
```
Comment (tarunnjoshi) [inline wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php:150]:
```suggestion
    if (!$subTypes) {
```

**Review by tarunnjoshi — APPROVED** [review]

File: wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php (hunk 5 above)
Comment (belwalshubham) [reply wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php:150]: But both of these things are the same. Why do we need to make a change like this?

**Pull request merged**